select.go

```diff
@@ -34,11 +35,22 @@ func selectChannel(coll *mgo.Collection, api *slack.Client) (slack.Channel, erro
 	}
 
 	// Every channel has already been announced. Truncate the collection then
-	// send a random one.
+	// send a random one that isn't in the 'ignored' collection.
 	log.Print("Couldn't find a new one")
 	if err := coll.DropCollection(); err != nil {
 		return slack.Channel{}, err
 	}
 
-	return all[rand.Intn(len(all))], nil
+	for len(all) > 0 {
```

Comment:
Won't this infinite loop if `ignore` == `all`? We don't shorten `all` when we skip.